app/move/app/view/index.js

```diff
@@ -18,13 +19,14 @@ const { localsActions, localsMoveDetails, localsTabs } = require('./middleware')
 router.get('/', (req, res) => res.redirect(`${req.baseUrl}/warnings`))
 
 // Define shared middleware
-router.use((req, res, next) => {
-  next()
-})
-router.use(setMove)
+
+// For all non-timeline routes use standard move middleware
+router.use(/\/((?!timeline).)*/, setMove)
+// For all timeline route use move events middleware
+router.use('/timeline', setMoveWithEvents)
```

Comment:
Out of curiosity, would this work if the routes were defined the other way around? As in, timeline first, so if it matches it does the thing, everything else after, removing the need for the regex? I seem to remember something about that in express routing. Not suggesting we do it, just curious.

Reply:
I don't think it would give us the behaviour we need. This is essentially acting as an `if/else`. So if the route contains `/timeline` run `setMoveWithEvents`, for all other routes run `setMove`. Both middleware set the `req.move` property to allow following middleware and controllers to use that property.

The reason for this was because the API request runs a little slower when we ask for all timeline events so we had to split them as separate calls.

If we swapped them around and removed `/\/((?!timeline).)*/` we'd get both of them being run which would mean the last call would override the `req.move` property and leave it without the `timeline_events` property which is what is needed for the timeline route.